src/modules/generatePopupMarkup.js

```diff
@@ -0,0 +1,61 @@
+const generatePopupMarkup = (data) => {
+  const { detailedCardData, commentsData } = data;
+
+  const commentsMarkup = (commentsData) => {
+    if (commentsData?.error) {
+      return `<p class="no-comments-yet">No comments yet.</p>`;
+    }
+
+    return commentsData
+      .map((comment) => {
+        return `<p>${comment.creation_date}, ${comment.username} : ${comment.comment}</p>`;
+      })
+      .join("");
+  };
+
+  const resolvedCommentsMarkup = commentsMarkup(commentsData);
+
+  return `
+    <div class="popup-wrapper">
+                <div class="popup">
+                    <div class="popup-image-container" style='background-image: url(${
+                      detailedCardData.strMealThumb
+                    });'>
+                        <span class="close-popup-btn">x</span>
+                    </div>
+                    <div class="content-container">
+                        <h2 class="popup-title">${detailedCardData.strMeal}</h2>
+                        <div class="popup-summary">
+                           <p><a target="__blank" href="${
+                             detailedCardData.strYoutube
+                           }">Watch on Youtube</a></p>
+                           <p>Category: ${detailedCardData.strCategory}</p>
+                           <p>Area: ${detailedCardData.strArea}</p>
+                           <p>Meal Id: ${detailedCardData.idMeal}</p>
+                        </div>
+                        <div class="comments-container">
+                            <h3>Comments(<span class="comments-count">${
+                              commentsData.length || 0
```

Comment:
- Kindly make sure to follow the guidelines in the card `Add comments counter - Student B` in this [issues](https://github.com/serengia/capstone-project--consuming-API/issues/5):
   - [ ] You need to implement the counting logic in a separate module. 
   - [ ] A counter function should look for specific DOM elements. 
   - [ ] A counter function should cover all the edge cases you can think about.

Reply:
Hello @gemmen29 . Thank you for your feedback. I have been struggling for many hours without success to make the changes you requested work. Also, after several attempts, I still do not clearly understand the process I need to follow to implement the changes. 
### My concern
Besides the above-highlighted issues I am facing, I feel the requested changes will go against the DRY and KISS principles. Why do I think so? In my case, I am already passing the comments array (from which I can get the length) in my markup. Having a separate function to calculate and insert the comments count for every item will be unnecessary repetition besides making it harder to read my code.
Kindly advice. I feel stuck.

Reply:
Hi @serengia You need to count items from your page DOM instead of getting it directly from the API. You can do this by counting the DOM elements that have something similar like for example: `Class name`

### Hint: you need to use the Document object for more information you can see the documentation from [here](https://developer.mozilla.org/en-US/docs/Web/API/Document).

Reply:
Thank you @gemmen29  for getting back. We've been able to implement the change.